File: plugin/shared/src/main/java/com/jonathanpotts/cartographytable/shared/models/ChunkModel.java
package com.jonathanpotts.cartographytable.shared.models;

import java.util.Map;

/**
 * Stores data used to process a chunk.
 */
public class ChunkModel {
    /**
     * A map of coordinates to block models (YXZ format).
     */
    public Map<Integer, Map<Integer, Map<Integer, BlockModel>>> blocks;
}

File: plugin/shared/src/main/kotlin/com/jonathanpotts/cartographytable/shared/models/ChunkModel.kt
package com.jonathanpotts.cartographytable.shared.models

import kotlinx.serialization.Serializable

/**
 * Stores data used to process a chunk.
 *
 * @property blocks A map of coordinates to block models (YXZ format).
 */
@Serializable
data class ChunkModel(
    var blocks: MutableMap<Int, MutableMap<Int, MutableMap<Int, BlockModel>>> =
        mutableMapOf()
)
